modules/connectors/ninja_van/karrio/providers/ninja_van/error.py

```diff
@@ -0,0 +1,24 @@
+"""Karrio Ninja Van error parser."""
+import typing
+import karrio.lib as lib
+import karrio.core.models as models
+import karrio.providers.ninja_van.utils as provider_utils
+
+
+def parse_error_response(
+    response: dict,
+    settings: provider_utils.Settings,
+    **kwargs,
+) -> typing.List[models.Message]:
+    errors: list = []  # compute the carrier error object list
```

Comment:
Based on the NinjaVan API, when there is an error, this is the format they return

```
{
  "error": {
    "code": "string",
    "request_id": "string",
    "title": "string",
    "message": "string",
    "details": [
      {
        "reason": "string",
        "field": "string",
        "message": "string"
      }
    ]
  }
}
```

So you want to write the logic of parsing that error and collecting the information for Karrio.

The Sendle integration is a good example for this. Checkout 
https://github.com/karrioapi/karrio/blob/12792cff650d63817206aa86298e6b8201f3dd6b/modules/connectors/sendle/karrio/providers/sendle/error.py#L8-L37

and the schema JSON file looks like this https://github.com/karrioapi/karrio/blob/12792cff650d63817206aa86298e6b8201f3dd6b/modules/connectors/sendle/schemas/error_responses.json#L1-L16